test/tbb/test_concurrent_queue.cpp

```diff
@@ -281,3 +282,63 @@ TEST_CASE("Test clear and dtor with TrackableItem") {
     test_tracking_dtors_on_clear<oneapi::tbb::concurrent_queue<TrackableItem>>();
     test_tracking_dtors_on_clear<oneapi::tbb::concurrent_bounded_queue<TrackableItem>>();
 }
+
+template<typename CQ, typename T>
+void test_queue_helper(){
+    int size = 5;
+    T vec_1(size, 0), vec_2(size, 0), vec_3(size, 0), vec_4(size, 0);
+    srand(static_cast<unsigned>(time(0)));
+    generate(vec_1.begin(), vec_1.end(), rand);
+    generate(vec_2.begin(), vec_2.end(), rand);
+    generate(vec_3.begin(), vec_3.end(), rand);
+    generate(vec_4.begin(), vec_4.end(), rand);
+    
+    CQ q1, q2;
+    CQ q4({vec_1, vec_2, vec_3});
+    CQ q3 = {vec_4, vec_2, vec_3};
+
+    q1 = q3;
+    q2 = std::move(q3);
+
+    CHECK(q1 != q4);
+    q1.swap(q4);
+    CHECK(q2 == q4);
+    CHECK(q2 == q3);
+}
+
+TEST_CASE("Test iterator queue"){
+    test_queue_helper<tbb::concurrent_queue<std::vector<int>>, std::vector<int>>();
+    test_queue_helper<tbb::concurrent_bounded_queue<std::vector<int>>, std::vector<int>>();
+}
+
+template <typename queue_type, typename allocator_type>
+void TestMoveQueue(){
```

Comment:
Ideally, we would need to test move support as it is done for other containers. 
For example, https://github.com/oneapi-src/oneTBB/blob/master/test/conformance/conformance_concurrent_unordered_map.cpp#:~:text=TEST_CASE(%22concurrent_unordered_map%20move,%7D

Reply:
I added the test. Because container's member naming is different I rewrote a small test. I think with these two tests I am covering all the cases that are tested in other containers.